### PR TITLE
add tutorials for classification and detection

### PR DESCRIPTION
**Pull request by rickycao-qy**

### CODE REVIEW

File: docs/tutorials/component-image-classification.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+> With CUDA, GPUs can be conveniently used for general purpose calculations (a bit like numerical calculations performed in the CPU, etc.). Before CUDA, GPUs were generally only used for graphics rendering (such as through OpenGL, DirectX).
+
+```json
+pipcook run image-classification.json --verbose --tuna
+```
+Often the model will converge at 10-20 epochs. Of course, it depends on the complexity of your dataset. Model convergence means that the loss (loss value) is low enough and the accuracy is high enough.
+
+Logs are as following:
+```json
```
Comment (yorkie) [inline docs/tutorials/component-image-classification.md:145]:
```suggestion
```
```

File: docs/tutorials/component-image-classification.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+187/187 [==============================] - 11s 59ms/step - loss: 0.0107 - accuracy: 0.9980 - val_loss: 0.3830 - val_accuracy: 0.8388
+...
+187/187 [==============================] - 11s 61ms/step - loss: 3.0090e-05 - accuracy: 1.0000 - val_loss: 1.5646e-08 - val_accuracy: 1.0000
+Epoch 14/15
+187/187 [==============================] - 11s 61ms/step - loss: 5.1657e-05 - accuracy: 1.0000 - val_loss: 1.9073e-08 - val_accuracy: 1.0000
+Epoch 15/15
+187/187 [==============================] - 11s 61ms/step - loss: 5.1657e-05 - accuracy: 1.0000 - val_loss: 1.9073e-08 - val_accuracy: 1.0000
+```
+After the training is completed, output will be generated in the current directory, which is a brand-new npm package, then we first install dependencies:
+```json
```
Comment (yorkie) [inline docs/tutorials/component-image-classification.md:160]: Ditto.

File: docs/tutorials/component-image-classification.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+cd output
+npm install
```
Comment (yorkie) [inline docs/tutorials/component-image-classification.md:162]: Prepend the $ for all bash lines.

File: docs/tutorials/component-image-classification.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+(async () => {
+  const v1 = await predict('./test.jpg');
+  console.log(v1); 
+  // [[0.1, 0.9, 0.05, 0.05]]
+})();
+```
+Note that the prediction result we give is the probability of each category. You can process this probability to the result you want.
+
+## Conclusion
+In this way, the component recognition task based on the image classification model is completed. After completing the pipeline in our example, if you are interested in such tasks, you can also start preparing your own dataset for training. We have already introduced the format of the dataset in detail in the data preparation chapter. You only need to follow the file directory to easily prepare the data that matches our image classification pipeline. <br />
```
Comment (yorkie) [inline docs/tutorials/component-image-classification.md:177]:
```suggestion
In this way, the component recognition task based on the image classification model is completed. After completing the pipeline in our example, if you are interested in such tasks, you can also start preparing your own dataset for training. We have already introduced the format of the dataset in detail in the data preparation chapter. You only need to follow the file directory to easily prepare the data that matches our image classification pipeline.
```

File: docs/tutorials/component-object-detection.md [hunk 5]
```diff
@@ -0,0 +1,272 @@
+# Use Pipcook to detect front-end components in pictures
```
Comment (yorkie) [inline docs/tutorials/component-object-detection.md:1]: How about this title?

```suggestion
# Detecting the UI components from a design draft
```

File: docs/zh-cn/tutorials/component-image-classification.md [hunk 6]
```diff
@@ -0,0 +1,173 @@
+# 使用 Pipcook 分类图片中的前端组件
```
Comment (yorkie) [inline docs/zh-cn/tutorials/component-image-classification.md:1]: We can remove the "Pipcook" from title because these tutorials are in Pipcook.

File: docs/zh-cn/tutorials/component-image-classification.md [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+
+## 背景
+您是否在前端业务中遇到过这样的场景：手中有一些图片，您想有一种自动的方式来识别这些图片究竟是一个什么前端组件，它是一个按钮，还是一个导航栏，又或是一个表格？这便是一个典型的图片分类任务。<br />
```
Comment (yorkie) [inline docs/zh-cn/tutorials/component-image-classification.md:4]:
```suggestion
您是否在前端业务中遇到过这样的场景：手中有一些图片，您想有一种自动的方式来识别这些图片究竟是一个什么前端组件，它是一个按钮，还是一个导航栏，又或是一个表格？这便是一个典型的图片分类任务。
```

File: docs/zh-cn/tutorials/component-image-classification.md [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+
+> CUDA，Compute Unified Device Architecture的简称，是由NVIDIA公司创立的基于他们公司生产的图形处理器GPUs（Graphics Processing Units,可以通俗的理解为显卡）的一个并行计算平台和编程模型。
+> 通过CUDA，GPUs可以很方便地被用来进行通用计算（有点像在CPU中进行的数值计算等等）。在没有CUDA之前，GPUs一般只用来进行图形渲染（如通过OpenGL，DirectX）。
+
+```
+pipcook run image-classification.json --verbose --tuna
+```
+往往模型在 10-20 个 epoch 时就会收敛，当然，这取决于您的数据集复杂度。模型收敛是指 loss (损失值) 已经足够低并且准确度已经足够高了，在这种情况，每一个 epoch 对于模型的表现得改变已经不明显了。<br />
+<br />具体日志如下：
+```json
```
Comment (yorkie) [inline docs/zh-cn/tutorials/component-image-classification.md:141]: Ditto.

File: docs/zh-cn/tutorials/component-image-classification.md [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+1. **@pipcook/plugins-image-classification-tensorflow-model-train **我们使用此插件来进行模型的评估，模型的评估是指在测试集上模型的表现效果，这是一个图片分类基于 TensorFlow 的通用插件，和上一阶段具体选择的模型无关
+
+
+[mobilenet](https://zhuanlan.zhihu.com/p/31551004) 是轻量级模型，可以在 cpu 上进行训练，如果使用的是 [resnet](https://zhuanlan.zhihu.com/p/31852747)，由于模型本身比较大，因此在执行这个 Pipeline 之前推荐在有 cuda 环境的 GPU 机器上运行
+
+> CUDA，Compute Unified Device Architecture的简称，是由NVIDIA公司创立的基于他们公司生产的图形处理器GPUs（Graphics Processing Units,可以通俗的理解为显卡）的一个并行计算平台和编程模型。
+> 通过CUDA，GPUs可以很方便地被用来进行通用计算（有点像在CPU中进行的数值计算等等）。在没有CUDA之前，GPUs一般只用来进行图形渲染（如通过OpenGL，DirectX）。
+
+```
+pipcook run image-classification.json --verbose --tuna
```
Comment (yorkie) [inline docs/zh-cn/tutorials/component-image-classification.md:137]:
```suggestion
$ pipcook run image-classification.json --tuna
```

`--version` is not necessary.

File: packages/sdk/package.json [hunk 10]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@pipcook/sdk",
-  "version": "1.0.3",
+  "version": "1.0.0",
```
Comment (yorkie) [inline packages/sdk/package.json:3]: Why do we downgrade the SDK version, is it a typo?

File: docs/zh-cn/tutorials/component-object-detection.md [hunk 11]
```diff
@@ -0,0 +1,259 @@
...
+```
+注意，给出的结果包含三个部分：
+
+- boxes: 此属性是一个数组，每个元素是另一个包含四个元素的数组，分别是 xmin, xmax, ymin, ymax
+- scores：此属性是一个数组，每个元素是对应的预测结果的置信度
+- classes：此属性是一个数组，每个元素是对应的预测出来的类别
+<a name="EHuq8"></a>
+## 制作自己的数据集
+看完上面的描述，你是否已经迫不及待想要用目标检测解决自己的问题了呢，要想制作自己的数据集，主要有以下几步
+<a name="95EYZ"></a>
```
Comment (yorkie) [inline docs/zh-cn/tutorials/component-object-detection.md:239]: What's html tag <a> for?

File: docs/zh-cn/tutorials/component-object-detection.md [hunk 12]
```diff
@@ -0,0 +1,259 @@
...
+  //   ]
+  // }
+})();
+```
+注意，给出的结果包含三个部分：
+
+- boxes: 此属性是一个数组，每个元素是另一个包含四个元素的数组，分别是 xmin, xmax, ymin, ymax
+- scores：此属性是一个数组，每个元素是对应的预测结果的置信度
+- classes：此属性是一个数组，每个元素是对应的预测出来的类别
+<a name="EHuq8"></a>
```
Comment (yorkie) [inline docs/zh-cn/tutorials/component-object-detection.md:236]: Ditto.

**Comment by rickycao-qy** [comment]
@yorkie Have fixed <a> label problems, <br> label problems, titles and bash code cells. Please have a check please

File: docs/tutorials/component-object-detection.md [hunk 13]
```diff
@@ -0,0 +1,276 @@
...
+
+Have you ever encountered such a scenario in the front-end business: there are some pictures in your hand, and you want an automatic way to identify which components are included in this picture, where these components are located in the picture, and which type components are of. This kind of task are generally called object detection in the field of deep learning.
+
+> Object detection refers to finding targets from a scene (picture), including two processes: where and what
+
+This kind of detection is very useful. For example, in the research of code generation from image, the front-end code is mainly composed of div, img and span. We can identify the shape, bitmap, and text position in the image, and then directly generate the corresponding codes.
+
+This tutorial will teach you how to train a model to do such a detection task.
+
+## Scenario
```
Comment (yorkie) [inline docs/tutorials/component-object-detection.md:13]: We need to keep a new blank after headers.

File: docs/tutorials/component-object-detection.md [hunk 14]
```diff
@@ -0,0 +1,276 @@
...
+- size: width and height of image
+- object:
+   - name: component category
+   - bndbox: position of component
+
+
+We have prepared such a data set, you can download it and check it out: [Download](http://ai-sample.oss-cn-hangzhou.aliyuncs.com/pipcook/datasets/component-recognition-detection/component-recognition-detection.zip)
+
+## Start Training
+After the dataset is ready, we can start training. Using Pipcook can be very convenient for object detection training. You only need to build the pipeline like this,
```
Comment (yorkie) [inline docs/tutorials/component-object-detection.md:151]: Ditto

File: docs/tutorials/component-object-detection.md [hunk 15]
```diff
@@ -0,0 +1,276 @@
...
+This step is easier to understand. To have your own training data, you need to find a way to collect enough training pictures. In this step, you don’t need to label your own pictures. You only need to mark the original pictures. it is good
+
+### Labelling
+There are many labeling tools now, you can use these labeling tools to mark which components are on your original picture, what the locations are and what types are of each component, Let's take [labelimg](https://github.com/tzutalin/labelImg) as example
+
+![image.png](https://gw.alicdn.com/tfs/TB1nB4lN4z1gK0jSZSgXXavwpXa-799-401.png)
+
+You can install the software from the official labelimg website above, and then follow the steps below:
+
+
```
Comment (yorkie) [inline docs/tutorials/component-object-detection.md:264]: Remove the unneeded line.

**Comment by yorkie** [comment]
Thank you @rickycao-qy, and may I ask you to check these tutorials again to make sure is there only 1 empty line after a header.

**Comment by rickycao-qy** [comment]
> Thank you @rickycao-qy, and may I ask you to check these tutorials again to make sure is there only 1 empty line after a header.

Have fixed lines problems and punctuation problems~~

File: docs/tutorials/component-object-detection.md [hunk 16]
```diff
@@ -0,0 +1,282 @@
...
+```js
+{
+  boxes: [
+    [83, 31, 146, 71],  // xmin, ymin, xmax, ymax
+    [210, 48, 256, 78],
+    [403, 30, 653, 72],
+    [717, 41, 966, 83]
+  ],
+  classes: [
+  	0, 1, 2, 2  // class index
```
Comment (yorkie) [inline docs/tutorials/component-object-detection.md:29]: Incorrect indents

File: docs/tutorials/component-object-detection.md [hunk 17]
```diff
@@ -0,0 +1,282 @@
...
+  ],
+  scores: [
+  	0.95, 0.93, 0.96, 0.99 // scores
```
Comment (yorkie) [inline docs/tutorials/component-object-detection.md:32]: Ditto

File: docs/tutorials/component-object-detection.md [hunk 18]
```diff
@@ -0,0 +1,282 @@
...
+  ]
+}
+
```
Comment (yorkie) [inline docs/tutorials/component-object-detection.md:35]: Unneeded empty line.

File: docs/tutorials/component-object-detection.md [hunk 19]
```diff
@@ -0,0 +1,282 @@
...
+```
+At the same time, we will generate a labelmap during training. Labelmap is a mapping relationship between the serial number and the actual type. This generation is mainly due to the fact that our classification name is text, but before entering the model, we need to convert the text into numbers. Here is a labelmap:
```
Comment (yorkie) [inline docs/tutorials/component-object-detection.md:37]:
```suggestion
At the same time, we will generate a `labelmap` during training. Labelmap is a mapping relationship between the serial number and the actual type. This generation is mainly due to the fact that our classification name is text, but before entering the model, we need to convert the text into numbers. Here is a labelmap:
```

File: docs/tutorials/component-object-detection.md [hunk 20]
```diff
@@ -0,0 +1,282 @@
...
+- test
+   - 1.jpg
+   - 1.xml
+   - 2.jpg
+   - 2.xml
+   - ...
+
+We need to divide our dataset into a training set (train), a validation set (validation) and a test set (test) according to a certain proportion. Among them, the training set is mainly used to train the model, and the validation set and the test set are used to evaluate the model. The validation set is mainly used to evaluate the model during the training process to facilitate viewing of the model's overfitting and convergence. The test set is used to perform an overall evaluation of the model after all training is completed.
+
+For each picture, Pascal Voc specifies an xml annotation file to record which components and the location of each component in this picture. A typical xml file content is:
```
Comment (yorkie) [inline docs/tutorials/component-object-detection.md:78]:
```suggestion
For each picture, Pascal Voc specifies an XML annotation file to record which components and the location of each component in this picture. A typical XML file content is:
```

File: docs/tutorials/component-object-detection.md [hunk 21]
```diff
@@ -0,0 +1,282 @@
...
+         <xmin>717.46875</xmin>
+         <ymin>41.828125</ymin>
+         <xmax>966.96875</xmax>
+         <ymax>83.328125</ymax>
+      </bndbox>
+   </object>
+</annotation>
+```
+
+This xml annotation file is mainly composed of the following parts：
```
Comment (yorkie) [inline docs/tutorials/component-object-detection.md:141]:
```suggestion
This XML annotation file is mainly composed of the following parts：
```

File: docs/tutorials/component-object-detection.md [hunk 22]
```diff
@@ -0,0 +1,282 @@
...
+
+- folder / filename: These two fields mainly define the image position and type corresponding to the annotation
+- size: width and height of image
+- object:
+   - name: component category
+   - bndbox: position of component
+
+We have prepared such a data set, you can download it and check it out: [Download](http://ai-sample.oss-cn-hangzhou.aliyuncs.com/pipcook/datasets/component-recognition-detection/component-recognition-detection.zip).
```
Comment (yorkie) [inline docs/tutorials/component-object-detection.md:149]:
```suggestion
We have prepared such a dataset, you can download it and check it out: [Download](http://ai-sample.oss-cn-hangzhou.aliyuncs.com/pipcook/datasets/component-recognition-detection/component-recognition-detection.zip).
```

File: docs/tutorials/component-object-detection.md [hunk 23]
```diff
@@ -0,0 +1,282 @@
...
+1. **@pipcook/plugins-coco-data-access** Now that we have downloaded the dataset, we need to connect the dataset into the format required by the subsequent model. Since the detectron2 framework used by our model requires the coco format, we use this plugin.
+1. **@pipcook/plugins-detectron-fasterrcnn-model-define** We built a faster rcnn model based on the detectron2 framework. This model has a very good performance in the accuracy of object detection
+1. **@pipcook/plugins-detectron-model-train** This plugin is used to start the training of all kinds of models built on detectron2. We set iteration to 100000. If your dataset is very complex, you need to increase the number of iterations.
+1. **@pipcook/plugins-detectron-model-evaluate** We use this plug-in to evaluate the training effect of the model. This plugin will be effective only if the test testset is provided, and finally the average precision of each category is given.
+
+Since the object detection model, especially the model of the rcnn family is very large, it needs to be trained on a machine prepared with nvidia GPU and cuda 10.1 environment:
+
+```shell
+$ pipcook run object-detection.json --verbose --tuna
```
Comment (yorkie) [inline docs/tutorials/component-object-detection.md:193]:
```suggestion
$ pipcook run object-detection.json --verbose
```

Maybe it's unnecessary to use tuna for English readers?

File: docs/tutorials/component-object-detection.md [hunk 24]
```diff
@@ -0,0 +1,282 @@
...
+[06/28 10:29:32 d2.utils.events]:  iter: 0  total_loss: 4.249  loss_cls: 2.198  loss_box_reg: 0.056  loss_rpn_cls: 0.711  loss_rpn_loc: 1.084  data_time: 0.1073  lr: 0.000000  
+...
+[06/28 12:28:32 d2.utils.events]:  iter: 100000  total_loss: 0.032 loss_cls: 0.122  loss_box_reg: 0.056  loss_rpn_cls: 0.711  loss_rpn_loc: 1.084  data_time: 0.1073  lr: 0.000000  
+```
+
+After the training is completed, output will be generated in the current directory, which is a brand-new npm package, then we first install dependencies:
+
+```shell
+$ cd output
+$ BOA_TUNA=1 npm install
```
Comment (yorkie) [inline docs/tutorials/component-object-detection.md:219]:
```suggestion
$ npm install
```

File: docs/tutorials/component-object-detection.md [hunk 25]
```diff
@@ -0,0 +1,282 @@
...
+```
+
+After installing the environment, we can start to predict:
+First install dependencies:
```
Comment (yorkie) [inline docs/tutorials/component-object-detection.md:223]: Is this needed?

File: docs/zh-cn/tutorials/component-image-classification.md (hunk 9 above)
Comment (rickycao-qy) [reply docs/zh-cn/tutorials/component-image-classification.md]: why do we want to remove --verbose

**Comment by rickycao-qy** [comment]
@yorkie have fixed accordingly

**Comment by rickycao-qy** [comment]
> The TOCs should be updated:
> 
> * https://github.com/alibaba/pipcook/blob/master/docs/_sidebar.md
> * https://github.com/alibaba/pipcook/blob/master/docs/zh-cn/_sidebar.md

done
